kratos/input_output/logger_table_output.cpp

```diff
@@ -24,79 +24,87 @@
-        for(auto& column_name : mColumnsNames){
-            mColumnsWidth.push_back(column_name.size() + 1);
-            column_name.erase(column_name.find_last_not_of(" ") + 1);
-        }
+  LoggerTableOutput::LoggerTableOutput(std::ostream& rOutputStream, std::vector<std::string> const& ColumnsNames) : LoggerOutput(rOutputStream), mCurrentColumnIndex(0), mColumnsNames(ColumnsNames) {
+    for(auto& column_name : mColumnsNames){
+      mColumnsWidth.push_back(column_name.size() + 1);
+      column_name.erase(column_name.find_last_not_of(" ") + 1);
     }
 
-	LoggerTableOutput::LoggerTableOutput(LoggerTableOutput const& Other) : LoggerOutput(Other), mColumnsNames(Other.mColumnsNames), mColumnsWidth(Other.mColumnsWidth) {
-
+    const DataCommunicator& r_communicator = DataCommunicator::GetDefault();
+    if (r_communicator.Rank() == 0) {
+      WriteHeader();
     }
+  }
 
-	std::string LoggerTableOutput::Info() const
-	{
-		return "LoggerTableOutput";
-    }
-    
-    void LoggerTableOutput::WriteHeader()
-    {
-        for(std::size_t i = 0 ; i <  mColumnsNames.size() ; i++)
-            std::cout << std::left << std::setw(mColumnsWidth[i]) << mColumnsNames[i];
-        
-        std::cout << std::endl;
-
-        for(std::size_t i = 0 ; i <  mColumnsNames.size() ; i++)
-            this->GetStream() << std::left << std::setw(mColumnsWidth[i]) << mColumnsNames[i];
-        
-        this->GetStream() << std::endl;
-    }
+  LoggerTableOutput::LoggerTableOutput(LoggerTableOutput const& Other) : LoggerOutput(Other), mColumnsNames(Other.mColumnsNames), mColumnsWidth(Other.mColumnsWidth) {
+  }
 
-    void LoggerTableOutput::WriteMessage(LoggerMessage const& TheMessage)
-    {
-      int column_index = -1;
-
-      auto message_severity = TheMessage.GetSeverity();
-      if (message_severity <= this->GetSeverity()) {
-        for (std::size_t i = 0; i < mColumnsNames.size(); i++) {
-          if (mColumnsNames[i] == TheMessage.GetLabel()) {
-            column_index = static_cast<int>(i);
-            break;
-          }
-        }
-      }
+  std::string LoggerTableOutput::Info() const
+  {
+    return "LoggerTableOutput";
+  }
+
+  void LoggerTableOutput::WriteHeader()
+  {
+    for(std::size_t i = 0 ; i <  mColumnsNames.size() ; i++)
+      std::cout << std::left << std::setw(mColumnsWidth[i]) << mColumnsNames[i];
+
+    std::cout << std::endl;
+
+    for(std::size_t i = 0 ; i <  mColumnsNames.size() ; i++)
+      this->GetStream() << std::left << std::setw(mColumnsWidth[i]) << mColumnsNames[i];
 
-      if (column_index >= 0) { // The label found in columns
-        MoveCursorToColumn(column_index);
-        auto message = TheMessage.GetMessage();
-        message.erase(message.find_last_not_of(" \n\t") + 1);
-        this->GetStream()  << std::left << std::setw(mColumnsWidth[column_index]) << message;
+    this->GetStream() << std::endl;
+    mHeaderIsWritten = true;
```

Comment:
Why you are adding the mHeaderIsWritten here in info? Shouldn't be in WriteHeader or constructor where you call it?

Reply:
github code folding is being funny, `mHeaderIsWritten` is set at the end of `WriteHeader`.